subprojects/core-api/src/main/java/org/gradle/api/file/SyncSpec.java

```diff
@@ -34,6 +35,7 @@ public interface SyncSpec extends CopySpec {
      * @return the filter defining the files to preserve
      */
     @Internal
+    @ToBeKeptEagerProperty(because = "Read-only nested like property")
     PatternFilterable getPreserve();
```

Comment:
❓ Mark as `@Nested` instead?

Reply:
As discussed, cannot be annoted with `@Nested` since it's `@Internal` and `PatternFilterable`